lib/rubygems/platform.rb

```diff
@@ -181,11 +181,20 @@ def ===(other)
       # version
       (
         (@os != "linux" && (@version.nil? || other.version.nil?)) ||
-        (@os == "linux" && (other.version == "gnu#{@version}" || other.version == "musl#{@version}" || @version == "gnu#{other.version}")) ||
+        (@os == "linux" && (normalized_linux_version == other.normalized_linux_version || ["musl#{@version}", "musleabi#{@version}", "musleabihf#{@version}"].include?(other.version))) ||
         @version == other.version
       )
   end
 
```

Comment:
```suggestion
# Until it can be removed, changes to this method must also be reflected in `bundler/lib/bundler/rubygems_ext.rb`
```

Reply:
Definitely, I don't like having to keep this duplicated but I couldn't find a nicer way to do the backport. A note will definitely help here! Also changes to `===` need to be duplicated in Bundler, so I'll add a note there too.